python/docs/source/user_guide/sql/python_udtf.rst

```diff
@@ -0,0 +1,216 @@
+            cleanup or finalization operations after the UDTF has finished processing
+            all rows. It can also be used to yield additional rows if needed.
+
+            Yields
+            ------
+            tuple
+                A tuple representing a single row in the UDTF result relation.
+                Yield this if you want to return additional rows during termination.
+
+            Examples
+            --------
+            >>> def terminate(self) -> Iterator[Any]:
+            >>>     yield "done", None
+            """
+            ...
+
+
+The return type of the UDTF defines the schema of the table it outputs. 
+It must be either a ``StructType`` or a DDL string representing a struct type.
+
+**Example UDTF Implementation:**
+
+Here is a simple example of a UDTF implementation:
+
+.. literalinclude:: ../../../../../examples/src/main/python/sql/udtf.py
+    :language: python
+    :lines: 36-40
+    :dedent: 4
+
+
+**Instantiating the UDTF:**
+
+To make use of the UDTF, you'll first need to instantiate it using the :func:`udtf` function:
+
+.. literalinclude:: ../../../../../examples/src/main/python/sql/udtf.py
+    :language: python
+    :lines: 42-55
+    :dedent: 4
+
+
+**Using `udtf` as a Decorator:**
+
+An alternative way for implementing a UDTF is to use :func:`udtf` as a decorator:
+
+.. literalinclude:: ../../../../../examples/src/main/python/sql/udtf.py
+    :language: python
+    :lines: 60-77
+    :dedent: 4
+
+For more detailed usage, please see :func:`udtf`.
+
+
+Registering and Using Python UDTFs in SQL
+-----------------------------------------
+
+Python UDTFs can also be registered and used in SQL queries.
+
+.. literalinclude:: ../../../../../examples/src/main/python/sql/udtf.py
+    :language: python
+    :lines: 82-116
+    :dedent: 4
+
+
+Arrow Optimization
+------------------
+Apache Arrow is an in-memory columnar data format used in Spark to efficiently transfer
+data between JVM and Python processes. Apache Arrow is disabled by default for Python UDTFs.
+
+To enable Arrow optimization, set ``spark.sql.execution.pythonUDTF.arrow.enabled`` to ``true``.
+Alternatively, you can specify the ``useArrow`` parameter when declaring the UDTF:
+
+.. literalinclude:: ../../../../../examples/src/main/python/sql/udtf.py
+    :language: python
+    :lines: 121-126
+    :dedent: 4
+
+
+For more details, please see `Apache Arrow in PySpark <../arrow_pandas.rst>`_.
+
+
+More Examples
+-------------
+
+A Python UDTF that expands date ranges into individual dates:
+
+.. literalinclude:: ../../../../../examples/src/main/python/sql/udtf.py
+    :language: python
+    :lines: 131-152
+    :dedent: 4
+
+
+A Python UDTF with ``__init__`` and ``terminate``:
+
+.. literalinclude:: ../../../../../examples/src/main/python/sql/udtf.py
+    :language: python
+    :lines: 157-176
+    :dedent: 4
+
+
+Advanced Featuress
+------------------
+
+TABLE input argument
```

Comment:
I would recommend to propose this as the primary way of passing relation arguments, rather than in the "additional features" section, since this syntax conforms to the SQL standard.

One way is to just swap the LATERAL syntax to this "advanced features" section instead.

Reply:
Moved. But we might need to improve it in the future ([SPARK-44746](https://issues.apache.org/jira/browse/SPARK-44746))